### PR TITLE
Update nsight systems

### PR DESCRIPTION
**Pull request by charleskawczynski**
This PR updates to the latest version of nsight-systems.

### CODE REVIEW

File: .buildkite/dss_scaling/pipeline.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 env:
-  JULIA_VERSION: "1.10.0"
+  JULIA_VERSION: "1.10.5"
```
Comment (nefrathenrici) [inline .buildkite/dss_scaling/pipeline.yml:2]: Is this env variable needed?

**Review by nefrathenrici — APPROVED** [review]
Thanks Charlie! Just one question

File: .buildkite/dss_scaling/pipeline.yml (hunk 1 above)
Comment (charleskawczynski) [reply .buildkite/dss_scaling/pipeline.yml:2]: Good question, I'm not sure. Regardless, I think I'd rather not leave it outdated 🤷🏻‍♂️

File: .buildkite/dss_scaling/pipeline.yml [hunk 2]
```diff
@@ -37,7 +37,7 @@ steps:
     steps:
     - label: ":computer: Shallow water DSS scaling - (1) processes"
       command:
-        - "module load cuda/11.3 nsight-systems/2022.2.1"
+        - "module load cuda/11.3 nsight-systems/2024.6.1"
```
[the remark this answers is not in the document]
Comment (charleskawczynski) [reply .buildkite/dss_scaling/pipeline.yml:40]: Ah, yeah, this pipeline is pretty stale. I'll make a few updates

File: .buildkite/dss_scaling/pipeline.yml (hunk 1 above)
Comment (charleskawczynski) [reply .buildkite/dss_scaling/pipeline.yml]: Actually, this pipeline is pretty old, I'm going to update a few things and use climacommon.

File: .buildkite/dss_scaling/pipeline.yml [hunk 3]
```diff
@@ -1,14 +1,13 @@
+agents:
+  queue: new-central
+  slurm_mem: 8G
+  modules: climacommon/2024_10_08
+
 env:
-  JULIA_VERSION: "1.10.0"
   MPICH_VERSION: "4.0.0"
   OPENMPI_VERSION: "4.1.1"
   CUDA_VERSION: "11.2"
```
Comment (Sbozzolo) [inline .buildkite/dss_scaling/pipeline.yml:9]:
```suggestion

```

**Review by Sbozzolo — APPROVED** [review]
Thank you for updating this!

**Pull request merged**